src/trace.h

```diff
@@ -21,31 +21,53 @@ struct git_trace_data {
 
 extern struct git_trace_data git_trace__data;
 
-GIT_INLINE(void) git_trace__write_fmt(
+GIT_INLINE(void) git_trace__vwrite_fmt(
 	git_trace_level_t level,
-	const char *fmt, ...)
+	const char *fmt, va_list ap)
 {
 	git_trace_cb callback = git_trace__data.callback;
 	git_buf message = GIT_BUF_INIT;
-	va_list ap;
 
-	va_start(ap, fmt);
 	git_buf_vprintf(&message, fmt, ap);
-	va_end(ap);
 
 	callback(level, git_buf_cstr(&message));
 
 	git_buf_dispose(&message);
 }
 
+GIT_INLINE(void) git_trace__write_fmt(
+	git_trace_level_t level,
+	const char *fmt, ...)
+{
+	va_list ap;
+
+	va_start(ap, fmt);
+	git_trace__vwrite_fmt(level, fmt, ap);
+	va_end(ap);
+}
+
 #define git_trace_level()		(git_trace__data.level)
+/* Varadic macros are a C99 feature */
+#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
```

Comment:
I'd say we should just drop this indirection.  At this point, `git_trace` isn't used within our own codebase, it only exists as a thing that allows you - a libgit2 contributor - to dig in to things within your code.

For background - originally, I wanted to make sure that this code was always inlined (since `inline` is just a hint) - to make sure that we would add as little overhead as possible when tracing was off.  The assumption being that we would ship a lot of tracing code within libgit2 itself.  We haven't - and I don't think that we have a lot of plans to, at least not yet.

Let's just make it a `GIT_INLINE` function always instead of trying to make it a macro if that's supported and an inline function if it's not.  This won't hurt anything today (anybody putting trace instrumentation into libgit2 is doing it because they have tracing on.  And we can always revisit if we find it to need optimization - that data will guide the optimization we do, I think.

Reply:
It actually *is* - `src/transports/httpclient.c`. If it was just user code only, I wouldn't have caught it :grimacing:

Reply:
Oh ha!  I used that when I was refactoring httpclient and forgot to pull it out.  Or I made a conscious decision to keep it.  I actually don't recall.  😁 

Either way, I think simpler is better and let's just use an inline function.  I went overboard with the macro magic in the first place and now we have an opportunity to right that wrong.  😁